### ISSUE TITLE
Signal on page moving

### ISSUE THREAD

**Issue by pyMan** [wagtail/wagtail#2728]
Hello.

The existing page_published and page_unpublished signals are really useful to purge pages and more related urls from a reverse proxy like Varnish Cache. But that's not enough.

I think there should be a "pre" and a "post" signal like that for page moves too.

When you move a (live) page from a parent to another or when you change the page order under the same parent, the cache about its page and the ex/new sibling ones should be purged too.

Actually I can achive this in this way:

```
class MyPage(Page):
    ....
    def move(self, target, pos = None):
       if self.live and self.get_parent() != target :
            page_unpublished.send(sender = self.specific_class, instance = self.specific)

        super(MyPage, self).move(target, pos = pos)

        if self.live :
            self = Page.objects.get(id = self.id)
            page_published.send(sender = self.specific_class, instance = self.specific, revision = self)
```

Anyway this works only when the page is changing parent, because when just changing the sort order among the same parent it's the wagtail's view "set_page_position" (in the core) who directly calls Page.move (not the specific one).

So what about modify that view to call the "specific" move or what about the pre/post move signals?

**Comment by kaedroho** [comment]
Moving a page also causes the URL of it's children to be updated, so there could potentially be a lot of updates to do. The same issue occurs when a page's slug is changed as well.

If a site has index page purging set up and/or multiple languages (#3024) there moving a page with many children could potentially create a lot of cache purging requests. I think we should investigate bulk purging (#3750) before we create too many ways to generate cache purge requests.

**Comment by lb-** [comment]
Resolved via #5169
